### ISSUE TITLE
Set all the FactoryGirl.create's to the respective instance variable to access form the spec

### ISSUE THREAD

**Issue by Charizard** [Codebrahma/factory_group#3]
For example consider the following usage of `FactoryGroup`

``` ruby
class User
  attr_accessor :name, :settings
end

FactoryGroup.define :user_group do
  user FactoryGirl.create(:user)
  settings user.settings
end
```

here if the FactoryGroup is called from a spec, the spec should be having a variable `user` and `settings`

Consider the following way in which the FactoryGroup can be used

``` ruby
# user_spec.rb
describe User do
  let(:user_group) { FactoryGroup.create :user_group }
  it "user should be an instance of User" do
    user_group.user.should be_an_instance_of described_class
  end
  it "settings should be an instance of User" do
    user_group.settings.should be_an_instance_of String
  end
end
```

**Comment by aalavandhan** [comment]
How will they use this while writing a spec. Can you put up an example.

**Comment by Charizard** [comment]
@nithinkrishna Updated the comment with an example spec.

**Comment by aalavandhan** [comment]
`let` lazy loads the the definition. By the example you've provided, unless `user_group` is used at least once the `FactoryGroup.create :user_group` will not be executed. Shouldn't it be defined in a `before` block instead?

**Comment by Charizard** [comment]
Sorry, my bad. For now let's change it to `let!`. It cannot be inside a `before` block, because all the scope of variables like `user` and `settings` being set by FactoryGroup will be within the `before` block in which it was created.

**Comment by aalavandhan** [comment]
This seems like you are restricting the usage to `let!`. 
Consider the following type of usage.

``` ruby
# user_spec.rb
describe User do
  let(:user_group) { FactoryGroup.create :user_group }
  before(:all){
    @admin_group = FactoryGroup.create :admin_group
  }
  it "user name should be 'admin' or 'user' depending of the type" do
    expect(user_group.user.name).to eq('user')
    expect(@admin_group.user.name).to eq('admin')
  end
end
```

Advantages of this usage over the suggested example.
1. It can be used in both `before` and `let` blocks.
2. Allows users to define multiple groups within a spec file, and use data from any of the defined groups.

What do you think? @Charizard

**Comment by Charizard** [comment]
@nithinkrishna Yeah, I was working on #10 and was thinking the same thing. Ok, let's go ahead and do as you suggested.

**Comment by Charizard** [comment]
@nithinkrishna Have updated the description.

**Comment by Charizard** [comment]
@nithinkrishna Now, I have fixed #10 so, now it will be easy for you. Please check. Also open a pull request requesting to resolve this issue once you are done. Thanks.

**Comment by Charizard** [comment]
@nithinkrishna I believe the current issue is solved in #10 already. Am I right?

**Comment by aalavandhan** [comment]
Yea. I guess it does. This can be closed.
